### PR TITLE
Remove openbabel dependency in qchem

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
This PR removes the `qchem` dependency to `openbabel` which was used for converting different molecular geometry formats to the `xyz` format. We can safely expect the users to perform such conversions. The PR also collects the general-purpose functions of `qml.qchem` in a new `structure` module. All the functions in the new `structure` module will be still accessible with `qml.qchem.function_name`.

**Description of the Change:**
The function  `read_structure` is modified to remove its dependency to `openbabel` . The functions `read_structure`, `active_space`, `excitations`, `hf_state` and `excitations_to_wires` are moved inside `structure`.

**Benefits:**
Removing `qchem` dependencies. 

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by obliviateandsurrender — APPROVED** [review]
Looks good to go.

**Comment by josh146** [comment]
@soranjh do you also need to remove openbabel from the various actions? E.g., 

- .github\workflows\tests.yml
- doc\development\guide\installation.rst
- doc\introduction\chemistry.rst
- docker\qchem.dockerfile
- qchem\install-dependencies.sh
- qchem\README.rst

**Comment by soranjh** [comment]
> @soranjh do you also need to remove openbabel from the various actions and docs? E.g.,

Thanks @josh146. Removed openbabel everywhere.

File: doc/development/guide/installation.rst [hunk 1]
```diff
@@ -75,20 +75,6 @@ are required:
 
     conda install psi4 psi4-rt -c psi4
 
-* `Open Babel <https://openbabel.org>`__ (optional but recommended to run the full test suite)
```
Comment (ixfoduap) [inline doc/development/guide/installation.rst:78]: Great work identifying that we should remove this!

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -101,6 +101,8 @@
     [(#2385)](https://github.com/PennyLaneAI/pennylane/pull/2385)
   - The 6-31G basis set is added to the qchem basis set repo
     [(#2372)](https://github.com/PennyLaneAI/pennylane/pull/2372)
+  - The dependency to openbabel is removed
```
Comment (ixfoduap) [inline doc/releases/changelog-dev.md:104]:
```suggestion
  - The dependency on openbabel is removed
```

**Review by ixfoduap — APPROVED** [review]

File: doc/development/guide/installation.rst (hunk 1 above)
Comment (soranjh) [reply doc/development/guide/installation.rst:78]: Credit goes to @josh146.

File: docker/qchem.dockerfile [hunk 3]
```diff
@@ -17,7 +17,6 @@ FROM pennylane/base:latest
 # Update and install Qchem
 RUN DEBIAN_FRONTEND="noninteractive" apt-get install tzdata # need to perform this again
 RUN apt-get update \
-    && apt-get -y install --no-install-recommends make git openbabel \
```
Comment (josh146) [inline docker/qchem.dockerfile:20]: @soranjh sorry I missed this one! But `openbabel` was only one of the packages being installed, do we still need `make` and `git`? If so, we will need to revert the line to
```docker
    && apt-get -y install --no-install-recommends make git \
```

Comment (soranjh) [reply docker/qchem.dockerfile:20]: Hey Josh. I am actually making my final PR to bring qchem inside pennylane and remove the need for installation. I can create a small PR just to retrieve this line during the few days transition period until that PR gets merged.

**Pull request merged**